conda/common/configuration.py

```diff
@@ -1609,3 +1609,76 @@ def typify_parameter(self, parameter_name, value, source):
 
     def get_descriptions(self):
         raise NotImplementedError()
+
+
+def get_plugin_config_data(
+    data: dict[Path, dict[str, RawParameter]],
+) -> dict[Path, dict[str, RawParameter]]:
+    """
+    This is used to move everything under the key "plugins" from the provided dictionary
+    to the top level of the returned dictionary. The returned dictionary is then passed
+    to :class:`PluginConfig`.
+    """
+    new_data = defaultdict(dict)
+
+    for source, config in data.items():
+        if plugin_data := config.get("plugins"):
+            plugin_data_value = plugin_data.value(None)
+
+            if not isinstance(plugin_data_value, frozendict):
```

Comment:
Would it be safer to check for `dict`?

Reply:
We actually want `collections.abc.Mapping` instead. Checking for `dict` here will fail.